cassandra/cqlengine/models.py

```diff
@@ -835,9 +838,13 @@ def _class_get_connection(cls):
     def _inst_get_connection(self):
         return self._connection or self.__connection__
 
+    def __getitem__(self, s: slice | int) -> M | list[M]:
```

Comment:
this is a bit more then adding types, this add new functionality.
I think it should come on it's own commit, with the reasoning why it's needed